src/VisualStudio/CSharp/Impl/CSharpPackage.cs

```diff
@@ -70,25 +70,23 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
 
                 this.RegisterService<ICSharpTempPECompilerService>(async ct =>
                 {
+                    var workspace = this.ComponentModel.GetService<VisualStudioWorkspace>();
                     await JoinableTaskFactory.SwitchToMainThreadAsync(ct);
-                    return new TempPECompilerService(this.Workspace.Services.GetService<IMetadataService>());
+                    return new TempPECompilerService(workspace.Services.GetService<IMetadataService>());
                 });
             }
             catch (Exception e) when (FatalError.ReportAndPropagateUnlessCanceled(e))
             {
             }
         }
 
-        protected override VisualStudioWorkspaceImpl CreateWorkspace()
-            => this.ComponentModel.GetService<VisualStudioWorkspaceImpl>();
-
         protected override async Task RegisterObjectBrowserLibraryManagerAsync(CancellationToken cancellationToken)
         {
             await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
             if (await GetServiceAsync(typeof(SVsObjectManager)).ConfigureAwait(true) is IVsObjectManager2 objectManager)
             {
-                _libraryManager = new ObjectBrowserLibraryManager(this, ComponentModel, Workspace);
+                _libraryManager = new ObjectBrowserLibraryManager(this, ComponentModel, this.ComponentModel.GetService<VisualStudioWorkspace>());
```

Comment:
This doesn't match the commit comment of "I put the fetches before the switch to the UI thread". I'm guessing this could never be the first reference to it?

Reply:
I was feeling lazy. You caught me. Fixing.